src/libraries/System.Private.CoreLib/src/System/SearchValues/SearchValues.cs

```diff
@@ -18,6 +19,27 @@ namespace System.Buffers
     /// </remarks>
     public static class SearchValues
     {
+        private static SearchValues<char>? _whiteSpaces;
+
+        internal static SearchValues<char> WhiteSpaces
+        {
+            get
+            {
+                if (_whiteSpaces is null)
+                {
+                    List<char> whiteSpaces = [];
+                    for (int i = 0; i <= char.MaxValue; i++)
```

Comment:
This is non-trivial overhead in startup. This should be drive by a table in `CharUnicodeInfo`, which is generated by tool.

Reply:
Agreed, since we're in the runtime we can precompute the set and ensure we have sufficient test coverage.
In this case I think it should be
```c#
"\t\n\v\f\r \u0085\u00A0\u1680\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200A\u2028\u2029\u202F\u205F\u3000"
```
and then use a static readonly instead of lazy-init.

Reply:
If we're precomputing this, the implementation you're gonna get for `.Contains(char)` is
https://github.com/dotnet/runtime/blob/d2cdcdd69391bbfde1a8fb4aa275a6cc393ca65b/src/libraries/System.Private.CoreLib/src/System/SearchValues/ProbabilisticMapState.cs#L83
We can measure if it makes sense to switch `char.IsWhiteSpace` to using `SearchValues.Contains` here (or even hardcoding the constants if it matters).